pkg/controller/gkenetworkparamset/gkenetworkparamset_controller.go

```diff
@@ -173,27 +210,104 @@ func (c *Controller) syncGKENetworkParamSet(ctx context.Context, key string) err
 		return nil
 	}
 
-	params := obj.(*networkv1.GKENetworkParamSet)
+	originalParams := obj.(*networkv1.GKENetworkParamSet)
+	params := originalParams.DeepCopy()
+
+	err = c.syncGNP(ctx, params)
```

Comment:
will do

Reply:
Are you still planning to do this?

Reply:
No. Upon scoping out the implementation it's going to require a lot more code or reflection on subfields. The built in condition updating code doesnt return whether the original slice was modified, so we'd need to write our own or use reflection anyway.

I don't think speed will be an issue here because of the limited amount of GNP/network objects that can be created anyway. Its a lot of added complexity for a premature optimization. Its something we can easily go back and change if it becomes an issue.

Reply:
just unrelated to this PR but related to the discussion, look how cool was this https://github.com/kubernetes/kubernetes/pull/108078 :)